frontend/src/pages/post/detail/DetailsPostPage.tsx

```diff
@@ -0,0 +1,45 @@
+import PostDetail from './PostDetail.tsx';
+import { useParams } from "react-router-dom";
+import { PostDTO } from "../../../dto/PostDTO.tsx";
+import { useEffect, useState } from "react";
+import { Post } from "../../../models/Post.tsx";
+import { getPostByIdService } from "../../../services/PostService.tsx";
+import { convertPostDTOToPost } from "../../../util/maper/convertPostDTOToPost.tsx";
+
+type DetailsPostPageProps = {
+    deletePost: (postId: string) => void;
+}
+
+export default function DetailsPostPage({ deletePost }: Readonly<DetailsPostPageProps>) {
+
+    const [post, setPost] = useState<Post>();
+    const { postId } = useParams<{ postId: string }>();
+
+    if (postId === undefined) {
+        throw new Error('Undefined id');
+    } else {
+        useEffect(() => {
+            const fetchPost = async () => {
+                try {
+                    const postDTO: PostDTO = await getPostByIdService(postId);
+                    const post: Post = convertPostDTOToPost(postId, postDTO);
+                    setPost(post);
+                } catch (error) {
+                    console.error('Error getting post by id:', error);
+                }
+            };
+            fetchPost().then(r => r);
+        }, [postId]);
+    }
```

Comment:
Du solltest/darfst React hooks nicht in conditions benutzen bzw ein React Hook muss bei jedem Render aufgerufen werden und darf nicht durch eine Kondition nur manchmal aufgerufen werden.

```suggestion
useEffect(() => {
        if(!postId) return
    async () => {
        try {
            const postDTO: PostDTO = await getPostByIdService(postId);
                const post: Post = convertPostDTOToPost(postId, postDTO);
                setPost(post);
            } catch (error) {
                console.error('Error getting post by id:', error);
            }
        };
    }, [postId]);

    if (postId === undefined) {
        throw new Error('Undefined id');
    } 
```

Reply:
Geändert!